### PR TITLE
CLOUD-2423 New apis integration

### PR DESCRIPTION
**Pull request by lzac**
- Enabled token auth in both clients using token param.
- DRF token auth management in request calls preserving standard basic auth.
- Enabled test mode in both clients using test param.
- Integrated requests-futures to allow async calls for enabled resources defining async_get in resource methods.

### CODE REVIEW

**Comment by lzac** [comment]
Clients has been renamed as before, in my opinion we could improve the code and merge the clients in a single one optimizing resources management and specifying the base url in each resource using base classes. Please let me know if you agree with that before merging.

**Comment by lzac** [comment]
This is ready to merged in master imho.
Please review it @jibbolo, thanks!

File: cloudacademy/__init__.py [hunk 1]
```diff
@@ -108,38 +129,28 @@ class CloudAcademy(object):
...
-class CloudAcademyLabs(object):
-    """docstring for CloudAcademy"""
-    
-    _baseurl = "https://cloudacademy.com/labs/api"
-    
+class CloudAcademyLabs(BaseClient):
+    """
+    New Cloud Academy laboratories APIs client
+    """
+    base_url = "https://stage-ui.cloudacademy.com/labs/api"
```
Comment (jibbolo) [inline cloudacademy/__init__.py]: This must be configured during the class initialization and not hardcoded.

File: cloudacademy/__init__.py [hunk 2]
```diff
@@ -108,38 +129,28 @@ class CloudAcademy(object):
...
+        'laboratories': resources.Laboratories,
+        'user_laboratories': resources.UserLaboratories,
+    }
+
+
+class CloudAcademyBakery(BaseClient):
+    """
+    New Cloud Academy generic APIs client.
+    """
+    base_url = "https://stage-ui.cloudacademy.com/api"
```
Comment (jibbolo) [inline cloudacademy/__init__.py]: same as above

File: setup.py [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-  description = 'Python client for CloudAcademy.com API',
-  author = 'Giacomo Marinangeli',
-  author_email = 'giacomo@cloudacademy.com',
-  url = 'https://github.com/cloudacademy/python-api-client',
-  download_url = 'https://github.com/cloudacademy/python-api-client/archive/0.2.tar.gz',
-  keywords = ['cloudacademy', 'cloud', 'academy', 'api','rest'],
-  classifiers = [],
+  name='cloudacademy',
+  packages=['cloudacademy'],
+  version='0.2',
```
Comment (jibbolo) [inline setup.py]: update version number

File: setup.py [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
+  description='Python client for CloudAcademy.com API',
+  author='Giacomo Marinangeli',
+  author_email='giacomo@cloudacademy.com',
+  url='https://github.com/cloudacademy/python-api-client',
+  download_url='https://github.com/cloudacademy/python-api-client/archive/0.2.tar.gz',
+  keywords=['cloudacademy', 'cloud', 'academy', 'api','rest'],
+  classifiers=[],
   install_requires=[
       'requests >= 2.0.1',
```
Comment (jibbolo) [inline setup.py]: update requirements to reflect requirements.txt changes

File: cloudacademy/__init__.py (hunk 1 above)
Comment (lzac) [inline cloudacademy/__init__.py]: I'd like to have a default value. I'm going to move at as default value for the parameter in the init method.

File: cloudacademy/__init__.py (hunk 2 above)
Comment (lzac) [inline cloudacademy/__init__.py]: Same as above for me too :)

File: setup.py (hunk 3 above)
Comment (lzac) [inline setup.py]: Done

File: setup.py (hunk 4 above)
Comment (lzac) [inline setup.py]: Done

File: cloudacademy/__init__.py (hunk 1 above)
Comment (jibbolo) [inline cloudacademy/__init__.py]: This is a public client for our API and it should work out of the box, so the default value must be https://cloudacademy.com and replaced if needed on initialization.

Comment (lzac) [inline cloudacademy/__init__.py]: Modified with production one by default @robertoturrin

**Pull request merged**